src/pr-info.ts

```diff
@@ -451,30 +419,36 @@ configSuspicious["tsconfig.json"] = (contents, oldText) => {
             noEmit: true,
             forceConsistentCasingInFileNames: true
         }
+    },
+    [ "files", "compilerOptions.paths", "compilerOptions.baseUrl", "compilerOptions.typeRoots" ]
+);
+
+function isRelativePath(path: string) {
+    return path.split(/\//).every(part => part.length > 0 && !part.match(/^\.+$|[\\\n\r]/));
+}
+
+// helper for json file testers: allow either a given "requiredForm", or any edits that get closer
+// to it, ignoring some keys (sub-values can be specified using dots).  The ignored properties are
+// in most cases checked elsewhere (dtslint), and in some cases they are irrelevant.
+function makeJsonCheckerFromCore(requiredForm: any, ignoredKeys: string[]) {
+    const diffFromReq = (text: string) => {
+        let json: any;
+        try { json = JSON.parse(text); } catch (e) { return "couldn't parse json"; }
+        // allow dotted keys in ignoredKeys
+        ignoredKeys.map(k => k.split(".")).forEach(keys =>
+            delete keys.slice(0, -1).reduce((a,b) => a?.[b], json)?.[keys[keys.length - 1]]);
```

Comment:
Probably I'd do, but personal taste/background :slightly_smiling_face:
```suggestion
        ignoredKeys.forEach(k => {
            const keys = k.split(".");
            delete keys.slice(0, -1).reduce((a,b) => a?.[b], json)?.[keys[keys.length - 1]];
        });
```

Reply:
As a possibly lame compromise, I broke the line there...